### PR TITLE
ci/nodejs: use node v10, prefer shell builtins in ci scripts

### PR DESCRIPTION
**Pull request by janlazo**
Having issues because I haven't caught up with tree-sitter setup. Not sure why `TREE_SITTER_DIR` has to be in the user's home but whatever.

### CODE REVIEW

**Review by justinmk — APPROVED** [review]

File: ci/build.ps1 [hunk 1]
```diff
@@ -98,27 +99,17 @@ npm.cmd install -g neovim
...
-
-$client.DownloadFile("https://codeload.github.com/tree-sitter/tree-sitter-c/zip/v0.15.2","c:\treesitter\tree_sitter_c.zip")
-Expand-Archive c:\treesitter\tree_sitter_c.zip -DestinationPath c:\treesitter\
-cd c:\treesitter\tree-sitter-c-0.15.2
-c:\treesitter\tree-sitter.exe test
+$client.DownloadFile("https://codeload.github.com/tree-sitter/tree-sitter-c/zip/v0.15.2",".\tree_sitter_c.zip")
+Expand-Archive .\tree_sitter_c.zip -DestinationPath .
+cd tree-sitter-c-0.15.2
+tree-sitter.cmd test
+Test-Path -PathType Leaf "$env:TREE_SITTER_DIR\bin\c.dll"
```
Comment (janlazo) [inline ci/build.ps1]: This passes but I don't get why the following happens:

```
[  ERROR   ] test/functional\lua\treesitter_spec.lua @ 39: treesitter API with C parser before_each
test\functional\helpers.lua:106: Error executing lua: Failed to load parser: uv_dlopen: C:\Users\appveyor\tree-sitter-build/bin/c.dll is not a valid Win32 application
```

File: ci/build.ps1 [hunk 2]
```diff
@@ -98,27 +99,17 @@ npm.cmd install -g neovim
...
-python -c "import gzip, shutil; f1,f2 = gzip.open('tree-sitter-cli.gz', 'rb'), open('tree-sitter.exe', 'wb'); shutil.copyfileobj(f1, f2); f2.close()"
-
-$client.DownloadFile("https://codeload.github.com/tree-sitter/tree-sitter-c/zip/v0.15.2","c:\treesitter\tree_sitter_c.zip")
-Expand-Archive c:\treesitter\tree_sitter_c.zip -DestinationPath c:\treesitter\
-cd c:\treesitter\tree-sitter-c-0.15.2
-c:\treesitter\tree-sitter.exe test
+$client.DownloadFile("https://codeload.github.com/tree-sitter/tree-sitter-c/zip/v0.15.2",".\tree_sitter_c.zip")
+Expand-Archive .\tree_sitter_c.zip -DestinationPath .
+cd tree-sitter-c-0.15.2
+tree-sitter.cmd test
```
Comment (justinmk) [inline ci/build.ps1]: I guess this .cmd is coming from the node package, can't find it in the zip from https://codeload.github.com/tree-sitter/tree-sitter-c/zip/v0.15.2 . Is there a reason not to use the .exe? Does invoking `.cmd` directly in powershell potentially have problems?

File: ci/build.ps1 (hunk 1 above)
Comment (justinmk) [reply ci/build.ps1]: - is `uv_dlopen` is being strict about slashes (unlikely)? No, a similar mixed-slashes path is constructed by `treesitter_spec.lua`, and it passes on master.
- does `c.dll` needs to be sibling to the `.exe` or some other files? I don't see any changes here which would result in different contents of `bin/`.

Perhaps it's worth trying the old `c:\treesitter` path.

Comment (janlazo) [reply ci/build.ps1]: https://github.com/neovim/neovim/pull/10124#issuecomment-504610136

It has to do with VM arch on Appveyor.

File: ci/build.ps1 (hunk 2 above)
Comment (janlazo) [reply ci/build.ps1]: nodejs creates shell shims for "executable" scripts so the user can run it in the shell. nodejs doesn't expose 3rd-party `.exe` in `PATH` by default aside from the shipped binaries (ie `node.exe`).

File: ci/build.ps1 [hunk 3]
```diff
@@ -1,5 +1,6 @@
-$ErrorActionPreference = 'stop'
-Set-PSDebug -Strict -Trace 1
```
Comment (janlazo) [inline ci/build.ps1]: I never found this useful. It dumps more noise than unix shells with `set -x`.

File: ci/build.ps1 [hunk 4]
```diff
@@ -1,5 +1,6 @@
...
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+$ProgressPreference = 'SilentlyContinue'
```
Comment (janlazo) [inline ci/build.ps1]: to speedup `Invoke-WebRequest`.

File: ci/build.ps1 [hunk 5]
```diff
@@ -98,27 +98,21 @@ npm.cmd install -g neovim
...
 
-#npm.cmd install -g tree-sitter-cli
-#npm.cmd link tree-sitter-cli
 
-mkdir c:\treesitter
-$env:TREE_SITTER_DIR = "c:\treesitter"
-#$env:PATH = "c:\treesitter;$env:PATH"
-$client = new-object System.Net.WebClient
-cd c:\treesitter
+$env:TREE_SITTER_DIR = $env:USERPROFILE + "\tree-sitter-build"
```
Comment (janlazo) [inline ci/build.ps1]: Why does tree-sitter have to be in `$HOME` or `%USERPROFILE` on Windows?

**Pull request merged**